cg/meta/workflow/rnafusion.py

```diff
@@ -0,0 +1,315 @@
+            self.get_verified_arguments(
+                case_id=case_id,
+                work_dir=work_dir,
+                resume=resume,
+                profile=profile,
+                with_tower=with_tower,
+                stub=stub,
+                input=input,
+                outdir=outdir,
+                genomes_base=genomes_base,
+                trim=trim,
+                fusioninspector_filter=fusioninspector_filter,
+                all=all,
+                pizzly=pizzly,
+                squid=squid,
+                starfusion=starfusion,
+                fusioncatcher=fusioncatcher,
+                arriba=arriba,
+            )
+        )
+        nextflow_options = self.__build_command_str(
+            self.get_verified_arguments_nextflow(
+                case_id=case_id,
+                log=log,
+            )
+        )
+        command = ["run", self.nfcore_pipeline_path]
+        parameters = (
+            nextflow_options
+            + ["-bg", "-q"]
+            + command
+            + options
+            + [
+                " > "
+                + str(self.get_case_path(case_id))
+                + "/"
+                + case_id
+                + "-stdout.log 2> "
+                + str(self.get_case_path(case_id))
+                + "/"
+                + case_id
+                + "-stdout.err  < /dev/null & "
+            ]
+        )
+        self.process.run_command(parameters=parameters, dry_run=dry_run)
+
+    def get_deliverables_file_path(self, case_id: str) -> Path:
+        """Returns a path where the rnafusion deliverables file for the case_id should be located.
+
+        (Optional) Checks if deliverables file exists
+        """
+        return Path(self.get_case_path(case_id), case_id + "_deliverables.yaml")
+
+    def get_template_deliverables_file(self, rnafusion_bundle_template: Path) -> pd.DataFrame:
+        """Read deliverables file template and output to dataframe format"""
+        return pd.read_csv(rnafusion_bundle_template)
+
+    def edit_template_deliverables_file(
+        self, case_id: str, deliverables_template: pd.DataFrame
+    ) -> pd.DataFrame:
+        """Replace PATHTOCASE and CASEID from template deliverables file to corresponding strings, add path_index column"""
+        edited_deliverables = deliverables_template.replace(
+            {"PATHTOCASE": str(self.get_case_path(case_id))}, regex=True
+        )
+        edited_deliverables = edited_deliverables.replace({"CASEID": case_id}, regex=True)
+        edited_deliverables = edited_deliverables["path_index"] = "~"
+        return edited_deliverables
+
+    def convert_deliverables_dataframe_to_dict(self, dataframe: pd.DataFrame):
+        """Convert deliverables dataframe to dict with dict values in list format  [{column -> value}, … , {column -> value}] with (orient="records"). Also reformat apostrophes in path_index column"""
+        return dataframe.to_dict(orient="records").replace("'~'", "~")
```

Comment:
Why do you not do the replace in the function: ´edit_template_deliverables_file´, it would just be another key-value pair, and make this one more pure (to_dict and orient)?

Reply:
I tried it as you suggested first, but the apostrophes only appear after the conversion to dict

Reply:
Ah, ok. 👍

Reply:
Have you considered converting to dict as the first thing to do after reading the content. And then start parsing the python dict instead of the dataframe. You could even validate it by using a pydantic model instead of a dict (bonus task) and let the model to the replacement and setting of the keys/values. Less dependency on an external lib and more clear , testable and reusable python code